backends/concrete-cpu/implementation/src/c_api/fheint.rs

```diff
@@ -0,0 +1,175 @@
+            noise_level: 0,
+            message_modulus: 0,
+            carry_modulus: 0,
+            ks_first: false,
+        };
+    }
+
+    /// Create a `Ciphertext` from the lwe and `self` metadata
+    fn ct_from_lwe(&self, lwe: LweCiphertext<Vec<u64>>) -> Ciphertext {
+        Ciphertext::new(
+            lwe,
+            Degree::new(self.degree),
+            // TODO: how to set noise here?
+            NoiseLevel::UNKNOWN,
+            MessageModulus(self.message_modulus),
+            CarryModulus(self.carry_modulus),
+            if self.ks_first {
+                PBSOrder::KeyswitchBootstrap
+            } else {
+                PBSOrder::BootstrapKeyswitch
+            },
+        )
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn concrete_cpu_tfhers_uint8_description(
+    serialized_data_ptr: *const u8,
+    serialized_data_len: usize,
+) -> TfhersFheIntDescription {
+    nounwind(|| {
+        // deserialize fheuint8
+        let mut serialized_data = Cursor::new(slice::from_raw_parts(
+            serialized_data_ptr,
+            serialized_data_len,
+        ));
+        let fheuint: FheUint8 = match bincode::deserialize_from(&mut serialized_data) {
+            Ok(value) => value,
+            Err(_) => {
+                return TfhersFheIntDescription::zero();
+            }
+        };
+        // get metadata from fheuint's ciphertext
+        let (radix, _) = fheuint.into_raw_parts();
+        let blocks = radix.blocks();
+        let ct = match blocks.first() {
+            Some(value) => &value.ct,
+            None => {
+                return TfhersFheIntDescription::zero();
+            }
+        };
+        return TfhersFheIntDescription {
+            width: 8,
+            is_signed: false,
+            lwe_size: ct.lwe_size().0,
+            n_cts: blocks.len(),
+            degree: blocks[0].degree.get(),
+            noise_level: blocks[0].noise_level().get(),
+            message_modulus: blocks[0].message_modulus.0,
+            carry_modulus: blocks[0].carry_modulus.0,
+            ks_first: blocks[0].pbs_order == PBSOrder::KeyswitchBootstrap,
+        };
+    })
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn concrete_cpu_tfhers_uint8_to_lwe_array(
+    serialized_data_ptr: *const u8,
+    serialized_data_len: usize,
+    lwe_vec_buffer: *mut u64,
+) -> i64 {
+    nounwind(|| {
+        // deserialize fheuint8
+        let mut serialized_data = Cursor::new(slice::from_raw_parts(
+            serialized_data_ptr,
+            serialized_data_len,
+        ));
+        let fheuint: FheUint8 = match bincode::deserialize_from(&mut serialized_data) {
+            Ok(value) => value,
+            Err(_) => {
+                return 1;
+            }
+        };
+        // collect LWEs from fheuint
+        let (radix, _) = fheuint.into_raw_parts();
+        let blocks = radix.blocks();
+        let cts: Vec<LweCiphertext<Vec<u64>>> = blocks.into_iter().map(|b| b.ct.clone()).collect();
```

Comment:
Why do you need to perform a deep copy here ? Can't you iterate directly on `blocks.iter()` line 123 ?

Reply:
Indeed, it was an unnecessary clone. I just fixed it